src/table/table-model.class.ts

```diff
@@ -522,8 +522,10 @@ export class TableModel implements PaginationModel {
 	 * @param value state to set all rows to. Defaults to `true`
 	 */
 	selectAll(value = true) {
-		for (let i = 0; i < this.rowsSelected.length; i++) {
-			this.selectRow(i, value);
+		if (this.data.length > 1) {
```

Comment:
Shouldn't it be `if (this.data.length > 0) {`? When the `length` is 1 you should still be able o select all

Reply:
If there is no data in the model, it sets a single item so length is 1 when it's empty:
```
set data(newData: TableItem[][]) {
		if (!newData || (Array.isArray(newData) && newData.length === 0) ) {
			newData = [[]];
		}
```